security/src/main/java/io/airlift/security/utils/CloseableGSSContext.java

```diff
@@ -0,0 +1,192 @@
+package io.airlift.security.utils;
+
+import com.google.common.base.Throwables;
+import com.google.common.collect.ImmutableMap;
+import com.sun.security.auth.module.Krb5LoginModule;
+import io.airlift.log.Logger;
+import io.airlift.security.exception.AuthenticationException;
+import org.ietf.jgss.GSSContext;
+import org.ietf.jgss.GSSCredential;
+import org.ietf.jgss.GSSException;
+import org.ietf.jgss.GSSManager;
+import org.ietf.jgss.GSSName;
+import org.ietf.jgss.Oid;
+
+import javax.security.auth.Subject;
+import javax.security.auth.login.AppConfigurationEntry;
+import javax.security.auth.login.Configuration;
+import javax.security.auth.login.LoginContext;
+import javax.security.auth.login.LoginException;
+
+import java.io.Closeable;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.UnknownHostException;
+import java.security.Principal;
+import java.security.PrivilegedAction;
+import java.util.Locale;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+import static javax.security.auth.login.AppConfigurationEntry.LoginModuleControlFlag.REQUIRED;
+
+public class CloseableGSSContext
+        implements Closeable
+{
+    private static final Logger log = Logger.get(CloseableGSSContext.class);
+
+    private final GSSContext context;
+    private final LoginContext loginContext;
+
+    public CloseableGSSContext(String serviceName, String krb5Keytab, String krb5ccPath, String krb5ClientPrincipal, URI serviceUri)
+    {
+        requireNonNull(serviceName, "serviceName is null");
+        requireNonNull(serviceUri, "serviceUri is null");
+        try {
+            loginContext = new LoginContext("", null, null, new Configuration()
+            {
+                @Override
+                public AppConfigurationEntry[] getAppConfigurationEntry(String name)
+                {
+                    ImmutableMap.Builder<String, String> optionsBuilder = ImmutableMap.builder();
+                    optionsBuilder.put("refreshKrb5Config", "true");
+                    optionsBuilder.put("doNotPrompt", "true");
+                    if (log.isDebugEnabled()) {
+                        optionsBuilder.put("debug", "true");
+                    }
+
+                    if (krb5Keytab != null) {
+                        optionsBuilder.put("useKeytab", "true");
+                        optionsBuilder.put("keyTab", "keyTab");
```

Comment:
is this right? Did you mean to set the `keyTab` property to the value of `krb5Keytab`?